build.gradle

```diff
@@ -97,3 +95,8 @@ idea.project.ipr.withXml { xml ->
 jdks {
     daemonTarget = 17
 }
+
+javaVersions {
+    libraryTarget = 21
```

Comment:
I think one of the tricky parts here is IntelliJ runs with 17 rather than 21. In the IntelliJ plugin, we're probably going to have fork out to a new java process to run the formatting in a 21 JDK (there is already some code that does this iirc but it's slow. This is probably the impetus to get us to always fork but persist the JDK for a bit to make it fast).

Reply:
This is the existing code that uses the configured project JDK for running the formatter: https://github.com/palantir/palantir-java-format/pull/562

But yeah, it's slow given it spins up a new JVM for each format command.